Riot/Modules/Room/NotificationSettings/SwiftUI/RoomNotificationSettingsView.swift

```diff
@@ -79,7 +85,7 @@ fileprivate extension RoomNotificationSettingsViewState {
 }
 
 extension RoomNotificationState: Identifiable {
-    var id: String { UUID().uuidString }
+    var id: Int { self.rawValue }
```

Comment:
Wrapping the picker item's `viewModel.process` call in an `withAnimation` block will show you that each time it's tapped, the views are being swapped out for new ones. Whereas, using a consistent identifier will result in the changes being nicely animated.

I *think* this means that on every change of the view model, using a UUID would be the equivalent of reloading the whole tableview, whereas with an `enum`'s `rawValue` it will just update the cells that changed. Maybe irrelevant here, but useful as the tableview grows.

Reply:
Yea that's one I was meant to go back to, felt it was a bad idea 😂. Was actually causing one of the problems I was seeing. Thanks for that.